app/src/main/java/com/github/sdpcoachme/database/CachingStore.kt

```diff
@@ -731,7 +731,7 @@ class CachingStore(private val wrappedDatabase: Database,
      */
     fun setCurrentEmail(email: String): CompletableFuture<Void> {
         currentEmail = email
-        return completedFuture(null)
+        return storeLocalData()
```

Comment:
I think here it would be great to set the email only once retrieving data from the disk has been completed. This would also avoid races, especially in tests if we are not careful, the email set using setCurrentEmail can then be overriden by the email retrieved asynchronously from the disk in another thread.
The code could be as followed:

return retrieveLocalData
    .thenApply { currentEmail = email }
    .thenCompose { storeLocalData() }

Reply:
A lot of methods in the CachingStore do not follow this pattern (most of them actually). To me, it would make sense for all of them to wait for retrieveLocalData before presenting their result (getUser, updateUser, etc.) to prevent races. Should we update them all ?

Reply:
Yes, we should update them all I agree with that... thanks.
Feel free to write it as an issue if you are not able to take care of this now :)

Reply:
I've added it as an issue #182 so that we can merge this ASAP.